### PR TITLE
chore(docs): add doc aliases

### LINKED ISSUE #833: doc(alias) for better intellisense
It seems like the main motivator here is intellisense so if we do something we could use doc(alias) for things that make sense for autocomplete, and keep the other type names the same.

_Originally posted by @gakonst in https://github.com/alloy-rs/alloy/issues/814#issuecomment-2150796624_

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Closes #833 

## Solution

Adds dual `Tx` <> `Transaction` doc alias bindings as a non-intrusive way to resolve inconsistent naming: https://github.com/alloy-rs/alloy/issues/814#issuecomment-2150796624

Related `alloy-core` PR: https://github.com/alloy-rs/core/pull/649

## PR Checklist

- [ ] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Review by mattsse — APPROVED** [review]

File: crates/consensus/src/signed.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@ use alloy_primitives::{Signature, B256};
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Signed<T, Sig = Signature> {
     #[cfg_attr(feature = "serde", serde(flatten))]
+    #[doc(alias = "transaction")]
     tx: T,
     #[cfg_attr(feature = "serde", serde(flatten))]
     signature: Sig,
```
Comment (DaniPopes) [inline crates/consensus/src/signed.rs:12]:
```suggestion
    signature: Sig,
    #[doc(alias = "tx_hash", alias = "transaction_hash")]
```

File: crates/consensus/src/transaction/eip2930.rs [hunk 2]
```diff
@@ -216,7 +217,8 @@ impl TxEip2930 {
         Ok(signed)
     }
 
-    /// Get transaction type.
+    /// Get transaction type
```
Comment (DaniPopes) [inline crates/consensus/src/transaction/eip2930.rs:220]:
```suggestion
    /// Get transaction type.
```

File: crates/consensus/src/transaction/eip4844.rs [hunk 3]
```diff
@@ -572,6 +577,7 @@ impl TxEip4844 {
     }
 
     /// Get transaction type
```
Comment (DaniPopes) [inline crates/consensus/src/transaction/eip4844.rs:579]:
```suggestion
    /// Get transaction type.
```

File: crates/consensus/src/transaction/envelope.rs [hunk 4]
```diff
@@ -164,6 +166,7 @@ impl TxEnvelope {
     }
 
     /// Return the hash of the inner Signed
```
Comment (DaniPopes) [inline crates/consensus/src/transaction/envelope.rs:168]:
```suggestion
    /// Return the hash of the inner Signed.
```

File: crates/network/src/ethereum/builder.rs [hunk 5]
```diff
@@ -147,10 +147,12 @@ impl TransactionBuilder<Ethereum> for TransactionRequest {
         common && (legacy || eip2930 || eip1559 || eip4844)
     }
 
+    #[doc(alias = "output_transaction_type")]
     fn output_tx_type(&self) -> TxType {
         self.preferred_type()
     }
 
+    #[doc(alias = "output_transaction_type")]
```
Comment (DaniPopes) [inline crates/network/src/ethereum/builder.rs:155]:
```suggestion
    #[doc(alias = "output_transaction_type_checked")]
```

File: crates/rpc-types-trace/src/common.rs [hunk 6]
```diff
@@ -12,21 +12,30 @@ pub enum TraceResult<Ok, Err> {
         /// Trace results produced by the tracer
         result: Ok,
         /// transaction hash
-        #[serde(skip_serializing_if = "Option::is_none", rename = "txHash")]
+        #[serde(
+            skip_serializing_if = "Option::is_none",
+            rename = "txHash",
+            alias = "transaction_hash"
+        )]
```
Comment (DaniPopes) [inline crates/rpc-types-trace/src/common.rs:19]: this is wrong

```suggestion
        #[serde(skip_serializing_if = "Option::is_none", rename = "txHash")]
        #[doc(alias = "transaction_hash")]
```

File: crates/rpc-types-trace/src/common.rs [hunk 7]
```diff
@@ -12,21 +12,30 @@ pub enum TraceResult<Ok, Err> {
...
     Error {
         /// Trace failure produced by the tracer
         error: Err,
         /// transaction hash
-        #[serde(skip_serializing_if = "Option::is_none", rename = "txHash")]
+        #[serde(
+            skip_serializing_if = "Option::is_none",
+            rename = "txHash",
+            alias = "transaction_hash"
+        )]
```
Comment (DaniPopes) [inline crates/rpc-types-trace/src/common.rs:31]:
```suggestion
        #[serde(skip_serializing_if = "Option::is_none", rename = "txHash")]
        #[doc(alias = "transaction_hash")]
```

**Review by DaniPopes — APPROVED** [review]

**Comment by zerosnacks** [comment]
Good catches and thanks for the fixes! @DaniPopes

**Pull request merged**